merkle/log_verifier.go

```diff
@@ -248,3 +248,101 @@ func (v LogVerifier) VerifiedPrefixHashFromInclusionProof(
 	}
 	return res, nil
 }
+
+// VerifyRangeInclusion verifies that [begin, end) range, consisting of the
+// |leaves| with the specified hashes, is included into the tree with the
+// specified |size| and |root| hash, based on inclusion proofs for entries
+// number |begin| and |end-1| respectively.
+//
+// TODO(pavelkalinnikov): Current implementation has some level of redundancy,
+// as the combination of 2 inclusion proofs is a superset of the minimal range
+// inclusion proof. Consider adding proper range inclusion proof notion to
+// Trillian's API.
+func (v LogVerifier) VerifyRangeInclusion(
+	begin, end, size int64,
+	proof1, proof2 [][]byte,
+	root []byte, leaves [][]byte,
```

Comment:
It might be simper to supply leaves via `map[int64][]byte`. It would remove 2 arguments from the function signature, make the method more general, and avoid off-by one errors lining up the indexes between begin,end, and the leaves slice.